### ISSUE TITLE
输入一张图片和两张图片inference结果不一样

### ISSUE THREAD

**Issue by June-Li** [ultralytics/ultralytics#19441]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

import os
from ultralytics import YOLO
import cv2

# Load a model

model = YOLO(
    "/workspace/JuneLi/bbtv/DataGenerator/Generate_det_rec/train_chrs_det/weights/a/train2/weights/best.pt"
)  # pretrained YOLO11n model

# Run batched inference on a list of images
out_path = "/workspace/JuneLi/bbtv/DataGenerator/Generate_det_rec/train_chrs_det/test_out/test/"
if not os.path.exists(out_path):
    os.makedirs(out_path)
base_path = "/workspace/JuneLi/bbtv/DataGenerator/Generate_det_rec/chrs_det_yolo_type/yolo/images/test/"
# filepath_list = [base_path + filename for filename in os.listdir(base_path)]
filepath_list = [base_path + "15-37.jpg"]

results = model(
    filepath_list,  # filepaths to images
    device=[6],  # cuda device, None for CPU
    imgsz=960,  # inference size (pixels)
    batch=1,  # batch size
    project=out_path,  # save results to project/name
    save_conf=False,
    conf=0.4,
)  # return a list of Results objects

# Process results list
for idx_0, result in enumerate(results):
    get_num = int(
        filepath_list[idx_0].split("/")[-1].split(".", 1)[0].split("-", 1)[-1]
    )
    boxes = result.boxes  # Boxes object for bounding box outputs
    # masks = result.masks  # Masks object for segmentation masks outputs
    # keypoints = result.keypoints  # Keypoints object for pose outputs
    # probs = result.probs  # Probs object for classification outputs
    # obb = result.obb  # Oriented boxes object for OBB outputs
    # result.show()  # display to screen
    # result.save(filename=out_path + filepath_list[idx_0].split("/")[-1])  # save to disk

    img = cv2.imread(filepath_list[idx_0])
    out_str = ""
    for box_ori in boxes[:get_num]:
        box = box_ori.xyxy[0].tolist()
        cv2.rectangle(
            img,
            (int(box[0]), int(box[1])),
            (int(box[2]), int(box[3])),
            (0, 255, 0),
            1,
        )
        out_str += str(box) + " " + str(box_ori.conf[0]) + "\n"
    cv2.imwrite(out_path + filepath_list[idx_0].split("/")[-1], img)
    open(out_path + filepath_list[idx_0].split("/")[-1] + ".txt", "w").write(out_str)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @June-Li, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com/) for guidance on using Ultralytics YOLO. There, you'll find detailed examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, which might help clarify your question.

If this is a 🐛 Bug Report related to the differing inference results between single and multiple image inputs, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/). This would help us debug effectively. You might also want to include additional details such as the shape and content of `15-37.jpg`, and confirm that the discrepancy is consistent across varying images or configurations.

If you're exploring training or inference ❓ Questions, kindly share more information, such as your training dataset setup, training logs, and any modifications to the default YOLO parameters. Review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure the best practices are being followed.

To explore and discuss other issues or topics, join the Ultralytics community!  
🎧 Real-time chat: [Discord](https://discord.com/invite/ultralytics)  
📚 In-depth discussions: [Discourse](https://community.ultralytics.com/)  
🤝 Explore more: [Subreddit](https://reddit.com/r/Ultralytics)  

## Upgrade

Make sure you are using the latest version of the `ultralytics` package by upgrading it in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) installed:

```bash
pip install -U ultralytics
```

## Environments

To rule out any environment-related issues, try running YOLO in one of the following verified environments, all pre-configured with dependencies (e.g., [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn)):

- **Notebooks** with free GPU access:  
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) Deep Learning VM  
- [AWS Deep Learning AMI](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)  
- [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/): <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
The [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) runs daily tests on all supported platforms. If this badge is green, the latest release has passed all integration tests.  

An Ultralytics engineer will review your issue shortly 😊! In the meantime, providing as much context as possible will help expedite the resolution.

**Comment by Y-T-G** [comment]
It's due to minimum rectangle resizing.

https://github.com/ultralytics/ultralytics/issues/16388#issuecomment-2363289302

https://github.com/ultralytics/ultralytics/issues/16339#issuecomment-2358362884

**Comment by June-Li** [comment]
> It's due to minimum rectangle resizing.
> 
> [#16388 (comment)](https://github.com/ultralytics/ultralytics/issues/16388#issuecomment-2363289302)
> 
> [#16339 (comment)](https://github.com/ultralytics/ultralytics/issues/16339#issuecomment-2358362884)

我应该怎么解决这个问题呢，但图片推理结果是错的，两张图片输入进去结果是对的，所以我现在就是造一张白色的图片和要处理的图片拼成batch=2，这样的话出来的结果就是对的了，除了这个办法还有其他解决方案吗？万分感谢！！！

**Comment by Y-T-G** [comment]
You can export model to TorchScript

**Comment by glenn-jocher** [comment]
Glad to help! For more stable inference across batch sizes, consider exporting your model with `nms=True` to embed non-max suppression directly in the graph. This can help maintain consistency between single and batched inputs. For implementation details, see the TorchScript Export documentation and Export Modes guide. 

Let us know if you need further clarification! 🚀
